### DISCUSSION TITLE
Help with finding "Parent" word of exceptions/contractions during tokenization

### DISCUSSION THREAD

**Discussion by irowberry** [explosion/spaCy#10725] [Help: Coding & Implementations] [answered]
I'm having an issue with the tokenization of contractions in English using spaCy. I know the normal behavior for a sentence like 
`"I've wondered about how she's doing."` results in `[I, 've, wondered, about, how, she, 's, doing, .]`

However, I want to be able to check the tokens and see that `[I, 've]` both came from the word `I've`, likewise that `[she, 's]` came from `she's`. 
Is there an attribute that already does this that I haven't seen? I've tried looking at the custom exceptions, but it only allows changing the attributes of `ORTH` and `NORM`.

I also saw that `nlp.tokenizer.explain(text)` returns
`[('SPECIAL-1', 'I'), ('SPECIAL-2', "'ve"), ('TOKEN', 'wondered'), ('TOKEN', 'about'), ('TOKEN', 'how'), ('SPECIAL-1', 'she'), ('SPECIAL-2', "'s"), ('TOKEN', 'doing'), ('SUFFIX', '.')]`

So I know that during the tokenization process, it's identifying exceptions, I just need to access them as an attribute of the tokens. Any help would be appreciated.

**Comment by adrianeboyd** [comment, marked as answer]
The main `Tokenizer` used in the pipeline doesn't track this information at all (it would affect the performance), so there's no way to access/store this.

`Tokenizer.explain` is a much slower implementation of the same algorithm just for debugging purposes.
